lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -661,47 +662,58 @@ restrictedTo (UTxO utxo) outs =
   in their corresponding block. This should be probably enough to cover for
   pretty much all our needs.
 
-  If slotting arithmetic has to be introduced, it will require proper thoughts.
+  The core wallet works with absolute 'SlotNo' values. The network node backends
+  convert 'SlotId' (epoch and local slot) values to simple 'SlotNo' values
+  using their configuration and chain state.
```

Comment:
I am still not convinced having the `NetworkLayer` be aware of the chain-state is a good idea. I think I am afraid of race-conditions between the following:
- `Wallet`/`Checkpoint` state
- `NetworkLayer` state (some kind of `MVar (Map Epoch BlockchainParameters)`?)
- Short rollbacks (that change blockchain parameters in the EBB)
- Long rollbacks (larger than one epoch)
- Multiple wallets restoring at different positions in the chain simultaneously